File: day23/coprocessor_conflag2.py
#!/usr/bin/env python3

MIN = 108100
MAX = 125100
STRIDE = 17

# sieve
isPrime = set()
for i in range(MAX+1):
    isPrime.add(i)

for i in range(2, MAX+1):
    if i in isPrime:
        for j in range(i*2, MAX+1, i):
            if j in isPrime:
                isPrime.remove(j)

cnt = 0
for i in range(MIN, MAX+1, STRIDE):
    if i not in isPrime:
        cnt += 1

print("result2:", cnt)

File: day23/coprocessor_conflag2.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

# sieve
my $MAX = 125100;
my @isPrime = map {1} (0..$MAX);
$isPrime[0] = $isPrime[1] = 0;

for my $i (2..$MAX) {
    if ($isPrime[$i]) {
        for (my $j = $i*2; $j <= $MAX; $j += $i) {
            $isPrime[$j] = 0;
        }
    }
}

my $cnt = 0;
for (my $i = 108100; $i <= 125100; $i += 17) {
    $cnt++ unless $isPrime[$i];
}

say "result2: $cnt";
